### ISSUE TITLE
To add Anime Data Analysis 

### ISSUE THREAD

**Issue by ayan-joshi** [abhisheks008/DL-Simplified#885]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Anime Data Analysis </br>
:red_circle: **Aim** :   To analyze the Anime Dataset using several parameters and then try to make predictions</br>
:red_circle: **Dataset** : 
[All_Anime.csv](https://github.com/user-attachments/files/16442474/All_Anime.csv)
</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Ayan Joshi 
- GitHub Profile Link : https://github.com/ayan-joshi
- Email ID : ayanjoshi182003@gmail.com
- Participant ID (if applicable): 
- Approach for this Project : So , I'll first try applying different algorithms then after comparing I'll add the implementation with the best algorithm with its results 
- What is your participant role? Yes I'm a part of GSSOC , but Its not only for the program I'm eager to contribute 

@abhisheks008 Assign the issue to me 

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by ayan-joshi** [comment]
@abhisheks008 Please have a look I want to contribute

**Comment by abhisheks008** [comment]
> @abhisheks008 Please have a look I want to contribute

I don't think there is any scope of implementing deep learning models for this dataset.

**Comment by ayan-joshi** [comment]
> > @abhisheks008 Please have a look I want to contribute
> 
> I don't think there is any scope of implementing deep learning models for this dataset.

@abhisheks008 I had one more topic with me 
Backpropagation in Neural Networks

should I change this issue to this hope this will match with deep learning ideas ?

**Comment by ayan-joshi** [comment]
> > > @abhisheks008 Please have a look I want to contribute
> > 
> > 
> > I don't think there is any scope of implementing deep learning models for this dataset.
> 
> @abhisheks008 I had one more topic with me Backpropagation in Neural Networks
> 
> should I change this issue to this hope this will match with deep learning ideas ?

@abhisheks008 Its related to my college project so it will be helpful from you if I get to contribute on this repo , We have a project to contribute to a open source project in ML and last year I contributed to this repo in GSSOC so without hesitating I chose DL simplified to contribute , I'll write a paper about contribution as well

**Comment by ayan-joshi** [comment]
@abhisheks008 or If u have any thing to add from ur side any issue that will be good as well

**Comment by abhisheks008** [comment]
Couldn't find any such problem statement behind this project idea.

**Comment by ayan-joshi** [comment]
> Couldn't find any such problem statement behind this project idea.

@abhisheks008 yes but this project will be a great addition to the repo or do u have any idea that should I add to the project that will be very helpful

**Comment by abhisheks008** [comment]
Can you elaborate the problem statement you are planning to solve here? Are you trying to showcase the implementation of backpropagation in neural networks? Which neural network models/architectures you are going to use here? 

Can you clarify on these, it'll be better to understand for me.

**Comment by ayan-joshi** [comment]
> Can you elaborate the problem statement you are planning to solve here? Are you trying to showcase the implementation of backpropagation in neural networks? Which neural network models/architectures you are going to use here?
> 
> Can you clarify on these, it'll be better to understand for me.

Backpropagation  algorithm is being used for training artificial neural network. It computes the gradient of the loss function with respect to each weight by the chain rule so this ultimately improves the performance of the neural network.

I'll use basic idea of neural network , first forward propagation then the mathematical function and at the end with backward propagation to complete the algorithm.

**Comment by abhisheks008** [comment]
> > Can you elaborate the problem statement you are planning to solve here? Are you trying to showcase the implementation of backpropagation in neural networks? Which neural network models/architectures you are going to use here?
> > Can you clarify on these, it'll be better to understand for me.
> 
> Backpropagation algorithm is being used for training artificial neural network. It computes the gradient of the loss function with respect to each weight by the chain rule so this ultimately improves the performance of the neural network.
> 
> I'll use basic idea of neural network , first forward propagation then the mathematical function and at the end with backward propagation to complete the algorithm.

Any dataset you are using for this project?

**Comment by SnehaGupta0206** [comment]
Full name : Sneha Gupta
GitHub Profile Link : https://github.com/[SnehaGupta0206](https://github.com/SnehaGupta0206)
Email ID : 02snehagupta06@gmail.com
Approach for this Project : So, I will try to implement backpropagation in neural networks using various optimization algorithms such as Gradient Descent, SGD, Mini-Batch Gradient Descent, and Adam Optimizer. I will perform EDA on the Iris dataset. I will compare the models using accuracy and loss metrics and visualize the results.
What is your participant role? 
I am participating in GSSoC 2026 and I am excited to contribute to open-source projects.
@abhisheks008 Assign the issue to me

**Comment by abhisheks008** [comment]
Hi @SnehaGupta0206 can you please elaborate the approach

**Comment by SnehaGupta0206** [comment]
Thank you for your reply! @abhisheks008 

My Approach is as follows:
1.Data Cleaning and Preprocessing: where I will be using Iris dataset and during normalization, train-test split to make data to implement the neural network model
2.Exploratory Data Analysis: where I would analyze the structure, class distribution, correlations and features relationship and visualize using the graphs. 
3. Implementation: implement neural network for classification and apply backpropagation for weight updates during training.
4. Comparison in Algorithms: I will compare 4 algorithms that I have mentioned above.
5. Evaluation: I will compare the models using confusion matrix and accuracy.
6. Documentation

**Comment by abhisheks008** [comment]
> Thank you for your reply! @abhisheks008 
> 
> My Approach is as follows:
> 1.Data Cleaning and Preprocessing: where I will be using Iris dataset and during normalization, train-test split to make data to implement the neural network model
> 2.Exploratory Data Analysis: where I would analyze the structure, class distribution, correlations and features relationship and visualize using the graphs. 
> 3. Implementation: implement neural network for classification and apply backpropagation for weight updates during training.
> 4. Comparison in Algorithms: I will compare 4 algorithms that I have mentioned above.
> 5. Evaluation: I will compare the models using confusion matrix and accuracy.
> 6. Documentation

Hi @SnehaGupta0206 

- Have you finalized the dataset? If yes, can you share the same 
- What are the 4 models you have finalized? Can you please share them? And why those 4 models?
- On what basis the models will be suitable with the dataset?
- How will you check the accuracy and performance of the models?

**Comment by SnehaGupta0206** [comment]
Hi @abhisheks008 

Dataset:
I am planning to use Iris Dataset which is In-Built dataset because it is suitable for classification tasks, lightweight and easy to visualize. it is multiclass classification dataset with numerical features.

Models/Algorithms:
I am planning to compare Gradient Descent, Stochastic Gradient Descent, Mini-Batch Gradient Descent, and Adam Optimizer because they are fundamental optimization techniques used in neural networks. Adam generally provides faster convergence, while SGD and Mini-Batch help in understanding learning behavior comparatively.
Dataset is suitable for this as it is numerical features dataset which make it easier to implements the optimization techniques without requiring any high computational resources.

I will evaluate model performance using: Accuracy score, Confusion Matrix, Training Vs Validation Accuracy Graphs.

This comparison will help analyze the prediction accuracy, convergence speed and overall model performance .

**Comment by abhisheks008** [comment]
> Hi @abhisheks008 
> 
> Dataset:
> I am planning to use Iris Dataset which is In-Built dataset because it is suitable for classification tasks, lightweight and easy to visualize. it is multiclass classification dataset with numerical features.
> 
> Models/Algorithms:
> I am planning to compare Gradient Descent, Stochastic Gradient Descent, Mini-Batch Gradient Descent, and Adam Optimizer because they are fundamental optimization techniques used in neural networks. Adam generally provides faster convergence, while SGD and Mini-Batch help in understanding learning behavior comparatively.
> Dataset is suitable for this as it is numerical features dataset which make it easier to implements the optimization techniques without requiring any high computational resources.
> 
> I will evaluate model performance using: Accuracy score, Confusion Matrix, Training Vs Validation Accuracy Graphs.
> 
> This comparison will help analyze the prediction accuracy, convergence speed and overall model performance .
> 

Hi @SnehaGupta0206 issue assigned to you. You can start working on it.
